src/benchmarks/micro/corefx/System.Text.Json/Utf8JsonReader/Perf.Segment.cs

```diff
@@ -0,0 +1,142 @@
+        public TestCaseType TestCase;
+
+        public static IEnumerable<TestCaseType> TestCaseValues() => (IEnumerable<TestCaseType>)Enum.GetValues(typeof(TestCaseType));
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            _jsonString = JsonStrings.ResourceManager.GetString(TestCase.ToString());
+
+            _dataUtf8 = Encoding.UTF8.GetBytes(_jsonString);
+
+            _sequenceSingle = new ReadOnlySequence<byte>(_dataUtf8);
+
+            int[] segmentSizes = { 4_096, 8_192 };
+
+            _sequences = new Dictionary<int, ReadOnlySequence<byte>>();
+
+            for (int i = 0; i < segmentSizes.Length; i++)
+            {
+                int segmentSize = segmentSizes[i];
+                _sequences.Add(segmentSize, Utf8JsonReaderCommentsTests.GetSequence(_dataUtf8, segmentSize));
+            }
+        }
+
+        [Benchmark]
+        public void ReadSingleSegmentSequence()
+        {
+            var json = new Utf8JsonReader(_sequenceSingle);
+            while (json.Read()) ;
+        }
+
+        [Benchmark]
+        [Arguments(4_096)]
+        [Arguments(8_192)]
+        public void ReadSingleSegmentSequenceByN(int numberOfBytes)
+        {
+            int consumed = 0;
+            bool isFinalBlock = numberOfBytes >= _dataUtf8.Length;
+            JsonReaderState jsonState = default;
+            ReadOnlySpan<byte> fullDataAsSpan = _dataUtf8.AsSpan();
+
+            while (consumed < _dataUtf8.Length)
+            {
+                ReadOnlySpan<byte>  data = isFinalBlock?
+                    fullDataAsSpan.Slice(consumed) :
+                    fullDataAsSpan.Slice(consumed, numberOfBytes);
+
+                var json = new Utf8JsonReader(data, isFinalBlock, jsonState);
+
+                while (json.Read()) ;
+
+                consumed += (int)json.BytesConsumed;
+                jsonState = json.CurrentState;
+
+                if (consumed >= _dataUtf8.Length - numberOfBytes)
+                {
+                    isFinalBlock = true;
+                }
+            }
+        }
+
+        [Benchmark]
+        [Arguments(4_096)]
+        [Arguments(8_192)]
+        public void ReadMultiSegmentSequence(int segmentSize)
+        {
+            var json = new Utf8JsonReader(_sequences[segmentSize]);
+            while (json.Read()) ;
+        }
+
+        [Benchmark]
+        [Arguments(4_096)]
+        [Arguments(8_192)]
+        public void ReadMultiSegmentSequenceUsingSpan(int segmentSize)
+        {
+            ReadOnlySequence<byte> sequenceMultiple = _sequences[segmentSize];
+
+            byte[] buffer = ArrayPool<byte>.Shared.Rent(segmentSize * 2);
+            JsonReaderState state = default;
+            int previous = 0;
+            foreach (ReadOnlyMemory<byte> memory in sequenceMultiple)
+            {
+                ReadOnlySpan<byte> span = memory.Span;
+                Span<byte> bufferSpan = buffer;
+                span.CopyTo(bufferSpan.Slice(previous));
```

Comment:
what is the goal of performing this copy operation? is it not going to dominate the reported time?

Reply:
Perhaps is just to carryover the leftover of the current segment being read, I will check if there is a way to refactor this method.

Reply:
I ran vs profiler in the method and the results show `Read` dominating the execution time with 17ms.
![image](https://user-images.githubusercontent.com/16040868/63466923-2a097480-c419-11e9-90ed-5fd3b87f0e73.png)